src/openfermion/utils/_trotter_exp_to_qgates.py

```diff
@@ -0,0 +1,329 @@
+        for i in numpy.arange(cnot_pairs.shape[1])[::-1]:
+            pair = cnot_pairs[:, i]
+            cnots2.append("CNOT {} {}".format(pair[0], pair[1]))
+
+        # Exponentiating each Pauli string requires five parts
+
+        # 1. Perform basis rotations
+        output_stream.write('\n'.join(string_basis_1))
+        if not len(string_basis_1) == 0:
+            output_stream.write('\n')
+
+        # 2. First set CNOTs
+        output_stream.write('\n'.join(cnots1))
+        output_stream.write('\n')
+
+        # 3. Rotation (Note kexp & Ntrot)
+        output_stream.write("Rz {} {}\n".format(term_coeff, qids[-1]))
+
+        # 4. Second set of CNOTs
+        output_stream.write('\n'.join(cnots2))
+        output_stream.write('\n')
+
+        # 5. Rotate back to Z basis
+        output_stream.write('\n'.join(string_basis_2))
+        if not len(string_basis_2) == 0:
+            output_stream.write('\n')
+
+
+def product_exp_pauli_strings_to_qasm(output_stream, op_list):
+    """Sends a list of QubitOperators to be exponentiated one by one
+    and printed to a QASM stream one at a time. The input is a list of
+    Pauli-strings and the output is the ordered product of their
+    exponents.
+
+    Args:
+    output_stream (file/string/IO stream): QASM output stream
+    op_list(list of QubitOperators): list of QubitOperators to be
+        exponentiated
+
+    Returns:
+        None
+    """
+
+    for op in op_list:
+        exp_sgl_pauli_string_to_qasm(output_stream, op)
+
+
+def trotterize_exp_qubop_to_qasm(
+                output_stream,
+                hamiltonian,
+                trotter_number=None,
+                trotter_order=None,
+                term_ordering=None,
+                k_exp=None):
+    """Trotterize a Qubit hamiltonian, exponentiate each trotterized term
+    and write it to a QASM output stream.
+
+        Params:
+            output_stream (file/string/IO stream): QASM output stream
+            hamiltonian(QubitOperator): hamiltonian
+            trotter_number (int): optional number of trotter steps (slices) for
+                trotterization as an integer - default = 1
+            trotter_order: optional order of trotterization as an integer -
+                default = 1
+            term_ordering (list of (tuples of tuples)): list of tuples
+                (QubitOperator terms dictionary keys) that specifies
+                order of terms when trotterizing
+            k_exp (int, float, double): optional exponential factor to all
+                terms when trotterizing
+
+        Returns:
+            None
+
+    """
+
+    trotterized_ham = get_trotterized_qubops(hamiltonian,
+                                             trotter_number,
+                                             trotter_order,
+                                             term_ordering,
+                                             k_exp)
+    output_stream.write(str(count_qubits(hamiltonian))+'\n')
```

Comment:
There is clearly some file format in mind here, would be good to document it in the function doc.